### ISSUE TITLE
Wrong ordering of key in X in mapping

### ISSUE THREAD

**Issue by tekumara** [zed-industries/zed#4916]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When viewing [toast.yml](https://github.com/seek-oss/aec/blob/d7c841a/toast.yml)

### Environment

Zed: v0.77.3 (stable)
OS: macOS 13.2.1
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="699" alt="CleanShot 2023-03-17 at 14 25 10@2x" src="https://user-images.githubusercontent.com/125105/225805116-a27bbab6-c324-4777-bf3f-ddb5981d588a.png">

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by mikeschinkel** [comment]
I can reproduce this, definitely.

**Comment by mikeschinkel** [comment]
The file where I am experiencing this can be found [here](https://raw.githubusercontent.com/OAI/tools.openapis.org/b8271f1cd012712b6941d850caed45df335b7807/src/_data/tools.yaml) and I here is a screenshot showing the what I see:

![Image](https://github.com/user-attachments/assets/908ffd80-d101-40ac-94ae-a4090f1a57c0)

Here is my current settings file while running Zed `0.177.7` (`ab9b3f1ba31a79e125b493fa35d73bf066c66074`)

```
{
  "features": {
    "edit_prediction_provider": "zed"
  },
  "ui_font_size": 16,
  "buffer_font_size": 16,
  "theme": {
    "mode": "system",
    "light": "One Light",
    "dark": "One Dark"
  },
  "soft_wrap": "editor_width",
  "autosave": {
    "after_delay": {
      "milliseconds": 1000
    }
  },
  "restore_on_startup": "last_session",
  "lsp": {
    "yaml-language-server": {
      "settings": {
        "yaml": {
          "keyOrdering": true,
          "format": {
            "singleQuote": true
          },
          "schemas": {
              // "http://json.schemastore.org/composer": ["/*"],
              // "../relative/path/schema.json": ["/config*.yaml"]
          }
        }
      }
    }
  }
}
```

**Comment by mikeschinkel** [comment]
> _"Your keys are out of order (source comes after name alphabetically)."_


Yeah, I am just using the source doc as it was provided so reordering is not really in the cards. 

> _"You can trigger the LSP code action (cmd-.) which presents "Fix key order for this map" which will resort your map."_

Imagine the pushback I would get from a PR asking to reorder 1500+ entries of a 12mb file! 😲

> _"If you don't want to enforce this remove "keyOrdering": true, from your settings."_

Perfect, thank you!

P.S. Something to consider for future — though I am speaking of a much larger project than just this one thing — but it would be great if Zed provided _"how to fix"_ information with error messages like this one.  

Evidently both Elm and Rust provided _"how to fix"_ info with their error messages and which has been hailed as an real boon to developer productivity.  It probably has eliminated lots of otherwise soon-to-be-closed GitHub issues, too. 🙂
